src/main/java/uk/gov/ch/service/update/impl/OverseasOverseasEntityDataServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.ch.service.update.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ch.exception.EntityEmailNotFoundException;
+import uk.gov.ch.model.update.OverseasEntityData;
+import uk.gov.ch.model.update.OverseasEntityDataJson;
+import uk.gov.ch.repository.update.OverseasEntityDataRepository;
+import uk.gov.ch.service.update.OverseasEntityDataService;
+import uk.gov.ch.util.ApiLogger;
+
+import java.util.HashMap;
+
+import static uk.gov.ch.util.Constants.NOT_FOUND_MESSAGE;
+import static uk.gov.ch.util.Constants.OVERSEAS_ENTITY_ID_KEY;
+
+@Service
+public class OverseasOverseasEntityDataServiceImpl implements OverseasEntityDataService {
+
+    @Autowired
+    private OverseasEntityDataRepository overseasEntityDataRepository;
+
+    @Override
+    public OverseasEntityDataJson getEntityEmail(String incorporationNumber) throws EntityEmailNotFoundException {
+
+        HashMap<String, Object> logMap = new HashMap<>();
+        logMap.put(OVERSEAS_ENTITY_ID_KEY, (Object) incorporationNumber);
+        ApiLogger.infoContext(incorporationNumber, "Calling database during Update process to retrieve entity email for company", logMap);
+
+        OverseasEntityDataJson overseasEntityDataJson = new OverseasEntityDataJson();
+        OverseasEntityData overseasEntityData = overseasEntityDataRepository.getOverseasEntityData(incorporationNumber);
+
+        if (overseasEntityData.getEmailAddress() == null) {
```

Comment:
We'll allow "" and handle that in the caller.

Reply:
Is the behaviour then different if an empty string rather than a null is returned?